psi4/driver/procrouting/response/scf_response.py

```diff
@@ -156,61 +159,88 @@ def cpscf_linear_response(wfn, *args, **kwargs):
+                v_b = core.triplet(Co[1], vectors[i], Cv[1], True, False, False)
+            vectors_ab.append(v_a)
+            vectors_ab.append(v_b)
+        vectors = vectors_ab
 
     # compute response vectors for each input vector
     params = [kwargs.pop("conv_tol", 1.e-5), kwargs.pop("max_iter", 10), kwargs.pop("print_lvl", 2)]
 
     responses = wfn.cphf_solve(vectors, *params)
-
-    # zip vectors, responses for easy access
-    vectors = {k: v for k, v in zip(vector_names, vectors)}
-    responses = {k: v for k, v in zip(vector_names, responses)}
-
-    # compute response values, format output
-    output = []
-    for prop in complete_dict:
-
-        # try to replicate the data structure of the input
-        if 'User' in prop['name']:
-            if prop['length'] == 0:
-                output.append(responses[prop['vector names'][0]])
+    
+    if restricted:
+        # zip vectors, responses for easy access
+        vectors = {k: v for k, v in zip(vector_names, vectors)}
+        responses = {k: v for k, v in zip(vector_names, responses)}
+
+        # compute response values, format output
+        output = []
+        for prop in complete_dict:
+            # try to replicate the data structure of the input
+            if 'User' in prop['name']:
+                if prop['length'] == 0:
+                    output.append(responses[prop['vector names'][0]])
+                else:
+                    buf = []
+                    for name in prop['vector names']:
+                        buf.append(responses[name])
+                    output.append(buf)
             else:
-                buf = []
-                for name in prop['vector names']:
-                    buf.append(responses[name])
+                names = prop['vector names']
+                dim = len(names)
+                buf = np.zeros((dim, dim))
+                for i, i_name in enumerate(names):
+                    for j, j_name in enumerate(names):
+                        buf[i, j] = -1.0 * vectors[i_name].vector_dot(responses[j_name])
                 output.append(buf)
-
-        else:
-            names = prop['vector names']
-            dim = len(names)
-
-            buf = np.zeros((dim, dim))
-
-            for i, i_name in enumerate(names):
-                for j, j_name in enumerate(names):
-                    buf[i, j] = -1.0 * vectors[i_name].vector_dot(responses[j_name])
-
-            output.append(buf)
+    else:
+        vectors_a = {k: v for k, v in zip(vector_names, vectors[::2])}
+        vectors_b = {k: v for k, v in zip(vector_names, vectors[1::2])}
+        responses_a = {k: v for k, v in zip(vector_names, responses[::2])}
+        responses_b = {k: v for k, v in zip(vector_names, responses[1::2])}
+        output = []
+        names = prop['vector names']
+        dim = len(names)
+        buf = np.zeros((dim, dim))
+        for i, i_name in enumerate(names):
+            for j, j_name in enumerate(names):
```

Comment:
`for i, (vector_a, response_a) in enumerate(zip(vectors_a, responses_a)):` and same with below line. The loop contents will need to change slightly.

Reply:
Mhm... I'm not sure I understand this comment correctly. The loop is over Cartesian components, so to speak, and `vectors_a` and `responses_a` are both dictionaries with the same keys, so zipping them and iterating over the identical keys is probably not what we want?

Reply:
On rereading the code, I was confused. I think this point depends on how we want the user to input vectors, so I'll leave it for now.